### ISSUE TITLE
grep::test_muptiple_pattern_files_multiple_input_files_line_number inconsistent, fails sometimes

### ISSUE THREAD

**Issue by jgarzik** [rustcoreutils/posixutils-rs#219]
Test output during failure:
```
---- grep::test_muptiple_pattern_files_multiple_input_files_line_number stdout ----
thread 'grep::test_muptiple_pattern_files_multiple_input_files_line_number' panicked at plib/src/testing.rs:55:10:
failed to write to stdin: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    grep::test_muptiple_pattern_files_multiple_input_files_line_number
```

Re-running failed CI job usually resolves issue.

**Comment by jgarzik** [comment]
Similar failures:
```

---- grep::test_duplicate_input_files_count stdout ----
thread 'grep::test_duplicate_input_files_count' panicked at plib/src/testing.rs:55:10:
failed to write to stdin: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- grep::test_multiple_input_files_line_number stdout ----
thread 'grep::test_multiple_input_files_line_number' panicked at plib/src/testing.rs:55:10:
failed to write to stdin: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }

```

**Comment by jgarzik** [comment]
Believed fixed by #233
